content/docs/demos/ingress_contour_demo.md

```diff
@@ -0,0 +1,273 @@
+access-control-allow-credentials: true
+x-envoy-upstream-service-time: 3
+vary: Accept-Encoding
+```
+
+### HTTP Ingress (mTLS and TLS)
+
+To proxy connections to TLS backends using HTTPS, the backend service must be annotated with the port as follows:
+```bash
+kubectl annotate service httpbin -n httpbin projectcontour.io/upstream-protocol.tls='14001' --overwrite
+```
+
+Next, we need to create an HTTPProxy configuration to use TLS proxying to the backend service, and providing a CA certificate to validate the server certificate presented by the backend service. For this to work, we need to first delegate to Contour the permission to read OSM's CA certificate secret from the OSM's namespace when referenced in the HTTPProxy configuration in the `httpbin` namespace. Refer to the [Upstream TLS section](https://projectcontour.io/docs/v1.18.0/config/upstream-tls/) to learn more about upstream certificate validation and when certificate delegation is necessary. In addition, we must create an IngressBackend resource that specifies HTTPS ingress traffic directed to the `httpbin` service must only accept traffic from a trusted client, osm-contour-envoy in the ingress edge proxy we deployed. OSM automatically provisioned a client certificate for the `osm-contour-envoy` ingress gateway with the Subject ALternative Name (SAN) `osm-contour-envoy.osm-system.cluster.local` during install, so the IngressBackend configuration needs to reference the same SAN for mTLS authentication between the `osm-contour-envoy` edge and the `httpbin` backend.
+
+Apply the configurations:
+```bash
+kubectl apply -f - <<EOF
+apiVersion: projectcontour.io/v1
+kind: TLSCertificateDelegation
+metadata:
+  name: ca-secret
+  namespace: "$osm_namespace"
+spec:
+  delegations:
+    - secretName: osm-ca-bundle
+      targetNamespaces:
+      - httpbin
+---
+apiVersion: projectcontour.io/v1
+kind: HTTPProxy
+metadata:
+  name: httpbin
+  namespace: httpbin
+spec:
+  virtualhost:
+    fqdn: httpbin.org
+  routes:
+  - services:
+    - name: httpbin
+      port: 14001
+      validation:
+        caSecret: "$osm_namespace/osm-ca-bundle"
+        subjectName: httpbin.httpbin.cluster.local
+---
+kind: IngressBackend
+apiVersion: policy.openservicemesh.io/v1alpha1
+metadata:
+  name: httpbin
+  namespace: httpbin
+spec:
+  backends:
+  - name: httpbin
+    port:
+      number: 14001
+      protocol: https
+    tls:
+      skipClientCertValidation: false # mTLS (defaults to false)
+  sources:
+  - kind: Service
+    namespace: "$osm_namespace"
+    name: osm-contour-envoy
+  - kind: AuthenticatedPrincipal
+    name: "osm-contour-envoy.$osm_namespace.cluster.local"
+EOF
+```
+
+Now, we expect external clients to be able to access the `httpbin` service for HTTP requests for the `Host:` header `httpbin.org` with HTTPS proxying over mTLS between the ingress gateway and service backend:
+```console
+$ curl -sI http://"$ingress_host":"$ingress_port"/get -H "Host: httpbin.org"
```

Comment:
should this be https ?

Reply:
No, this is the request from the user to the ingress gateway, not the request from ingress gateway to backend.